modules/sync/admin-sync.php

```diff
@@ -230,4 +245,42 @@ public function __call( $func, $args ) {
 			E_USER_ERROR
 		);
 	}
+
+	/**
+	 * Returns some data (`from_post_id` and `new_lang`) from the current request.
+	 *
+	 * @since 3.7
+	 *
+	 * @param array $post_data A post array.
+	 * @return array
+	 *
+	 * @phpstan-return array{}|array{from_post_id: int<0,max>, new_lang: string}
+	 */
+	public static function get_data_from_request( array $post_data ): array {
+		if ( ! isset( $GLOBALS['pagenow'], $_GET['_wpnonce'], $_GET['from_post'], $_GET['new_lang'], $_GET['post_type'], $post_data['post_type'] ) ) {
+			return array();
+		}
+
+		if ( 'post-new.php' !== $GLOBALS['pagenow'] ) {
+			return array();
+		}
+
+		if ( ! is_string( $_GET['_wpnonce'] ) || ! is_numeric( $_GET['from_post'] ) || ! is_string( $_GET['new_lang'] ) ) {
+			return array();
+		}
+
+		if ( $post_data['post_type'] !== $_GET['post_type'] || ! pll_is_translated_post_type( $post_data['post_type'] ) ) {
+			return array();
+		}
+
+		// Capability check already done in post-new.php.
+		if ( ! wp_verify_nonce( wp_unslash( $_GET['_wpnonce'] ), 'new-post-translation' ) ) {
+			return array();
+		}
```

Comment:
We must use `check_admin_referer()` instead. This is our nonce and thus we are expect to check it ourselves and stop the execution if someone is cheating so I believe that `wp_verify_nonce()` is not the best in this case.
```suggestion
		check_admin_referer( 'new-post-translation' );
```

Reply:
Done in [72716d8](https://github.com/polylang/polylang/pull/1532/commits/72716d857d47221b40bbd3e18b063d5c6fcd7e6d).